### PR TITLE
Copying and pasting in Grid fixes.

### LINKED ISSUE #10760: Table Editor widget: pasting
Follow up of #10293 

The copying data in the widget works exactly the same way as copying from Table Visualization. We should also allow pasting such data to the widget, as well as from external editors.

### LINKED ISSUE #11141: Copying cells from the table viz on a mac keyboard shortcut
Copying cells from the table viz on a mac keyboard shortcut should be command-C not Control-C.

Same for other keyboard shortcuts

image

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Fixes #10760 
Fixes #11141

[Screencast From 2024-10-17 13-51-45.webm](https://github.com/user-attachments/assets/78924f87-6bdb-4cb6-8f95-d9f97c63aea4)

This PR also changes the name of new columns to `Column #n` where n is index of column - this way it's easier to quickly create tables with non-conflicting column names.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- ~~[ ] The documentation has been updated, if necessary.~~
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- ~~[ ] Unit tests have been written where possible.~~
- ~~[ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.~~
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: app/gui/src/project-view/components/GraphEditor/widgets/WidgetTableEditor.vue [hunk 1]
```diff
@@ -161,6 +162,20 @@ function onRowDragEnd(event: RowDragEndEvent<RowData>) {
   }
 }
 
+// === Paste Handler ===
+
+function processDataFromClipboard({ data, api }: ProcessDataFromClipboardParams<RowData>) {
+  const focusedCell = api.getFocusedCell()
+  if (focusedCell === null) console.warn('Pasting while no cell is focused!')
```
Comment (kazcw) [inline app/gui/src/project-view/components/GraphEditor/widgets/WidgetTableEditor.vue:169]: Unless this is expected to be unreachable, maybe it should be a toast.

File: app/gui/src/project-view/components/GraphEditor/widgets/WidgetTableEditor.vue [hunk 2]
```diff
@@ -161,6 +162,20 @@ function onRowDragEnd(event: RowDragEndEvent<RowData>) {
...
+  else {
+    pasteFromClipboard(data, {
+      rowIndex: focusedCell?.rowIndex,
```
Comment (kazcw) [inline app/gui/src/project-view/components/GraphEditor/widgets/WidgetTableEditor.vue]:
```suggestion
      rowIndex: focusedCell.rowIndex,
```

File: app/gui/src/project-view/components/shared/AgGridTableView.vue [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+    name: 'Copy',
+    shortcut: gridBindings.bindings['copyCells'].humanReadable,
+    action: ({ api }) => api.copyToClipboard(),
+    icon: AGGRID_DEFAULT_COPY_ICON,
+  },
+  paste: {
+    name: 'Paste',
+    shortcut: gridBindings.bindings['pasteCells'].humanReadable,
+    action: ({ api }) => api.pasteFromClipboard(),
+    icon: AGGRID_DEFAULT_COPY_ICON,
```
Comment (kazcw) [inline app/gui/src/project-view/components/shared/AgGridTableView.vue:35]: Should these three `icon` fields all have the same value?

**Review by kazcw — APPROVED** [review]

File: app/gui/src/project-view/components/shared/AgGridTableView.vue (hunk 3 above)
Comment (farmaazon) [reply app/gui/src/project-view/components/shared/AgGridTableView.vue:35]: They are the same in the "original" menu items of AgGrid

File: app/gui/src/project-view/components/GraphEditor/widgets/WidgetTableEditor.vue (hunk 1 above)
Comment (farmaazon) [reply app/gui/src/project-view/components/GraphEditor/widgets/WidgetTableEditor.vue:169]: It _should_ be unreachable. Without cell focused, the paste shortcut handler should not be called.

**Pull request merged**